File: SceneRetrieval/python/Class/Envelope.py
# -*- coding:utf-8 -*-

from Class.Point import *


class Envelope(object):
    # 外包矩形的四个顶点

    rtPoint = Point()  # 右上
    lbPoint = Point()  # 左下

    def __str__(self):
        return "rt:" + "x:" + bytes(self.rtPoint.x) + " y:" + bytes(self.rtPoint.y) + " lb:" + "x:" + bytes(
            self.lbPoint.x) + " y:" + bytes(self.lbPoint.y)

File: SceneRetrieval/model/Envelope.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SceneRetrieval.model
{
    class Envelope
    {
        public Point rtPoint;
        public Point lbPoint;

        private double width = 0;
        private double height = 0;

        public double Width
        {
            get
            {
                return (rtPoint.x - lbPoint.x) / 2;
            }

            set
            {
                width = value;
            }
        }

        public double Height
        {
            get
            {
                return (rtPoint.y - lbPoint.y) / 2;
            }

            set
            {
                height = value;
            }
        }
    }
}
